packages/@react-spectrum/card/src/WaterfallLayout.tsx

```diff
@@ -172,10 +172,9 @@ export class WaterfallLayout<T> extends BaseLayout<T> implements KeyboardDelegat
       // TODO: also not sure about copying layout info vs mutating it. Listlayout does the below
       // but I feel that is because it actually maintained a layoutNode map cache which this doesn't have
       let newLayoutInfo = layoutInfo.copy();
-      newLayoutInfo.rect.height = size.height;
+      newLayoutInfo.rect.height = size.height <= 600 ? size.height : 600;
```

Comment:
```suggestion
      newLayoutInfo.rect.height = size.height < 600 ? size.height : 600;
```

Reply:
I wanna keep the todos for now since they are open questions